src/main/java/com/hierynomus/protocol/commons/concurrent/TaskQueue.java

```diff
@@ -0,0 +1,120 @@
+/*
```

Comment:
Let's not copy-paste this. The license is not correct now anymore (wrong attribution).
Why aren't we using a simple `LinkedBlockingQueue` to queue the events?

Reply:
I see your point. Let me check how to implement a "TaskQueue" with a `LinkedBlockingQueue`.